src/Model/FsTreeNode.re

```diff
@@ -1,5 +1,9 @@
+open Oni_Core;
+
+module ArrayEx = Utility.ArrayEx;
+module Path = Utility.Path;
+
 type t = {
-  id: int,
```

Comment:
I've removed `id` entirely in favor of using paths as identifiers instead. For a few reasons:

- it's simpler.
- Paths don't change (and if they do it's because their identity changes). No more stale ids, they're valid across workspace changes and can even be persisted.
- Searching by path is most likely faster since nodes that aren't in the path can be eliminated from the search. Hashing also makes individual comparisons as fast as on ids.

Reply:
Makes sense to me!